sdk/storage/azblob/hashing/hashing_reader.go

```diff
@@ -0,0 +1,46 @@
+package hashing
+
+import (
+	"bytes"
+	"hash/crc64"
+	"io"
+)
+
+// ReadWrapper is an io.ReadSeekCloser wrapper to prevent reading twice for SDK-generated hashes.
+type ReadWrapper struct {
+	internal *bytes.Reader
+	crc64    uint64
+}
+
+func (h *ReadWrapper) CRC64Hash() uint64 {
+	return h.crc64
+}
+
+func (h *ReadWrapper) Read(p []byte) (n int, err error) {
+	return h.internal.Read(p)
+}
+
+func (h *ReadWrapper) Seek(offset int64, whence int) (int64, error) {
+	return h.internal.Seek(offset, whence)
+}
+
+func (h *ReadWrapper) Close() error {
+	return nil
+}
+
+func NewHashingReadWrapper(r io.ReadSeekCloser, validationOption StorageTransferValidationOption) (*ReadWrapper, error) {
```

Comment:
Since this is in the `hashing` package, you can omit the word "Hash" from the name.